src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -186,7 +192,17 @@ static void wrapped_rr_closure(grpc_exec_ctx *exec_ctx, void *arg,
 
   grpc_exec_ctx_sched(exec_ctx, wc_arg->wrapped_closure, GRPC_ERROR_REF(error),
                       NULL);
-  gpr_free(wc_arg->owning_pending_node);
+
+  /* Make sure this closure and its arg are EITHER on the heap on their oen OR
+   * part of a pending pick (thus part of the pending pick's memory) */
+  GPR_ASSERT((wc_arg->closure_mem_or_null != NULL) +
```

Comment:
Good idea, done.

We still need to do a free, even after taking your suggestion.